src/coredump/coredumpctl.c

```diff
@@ -1150,6 +1236,31 @@ static int run_debug(int argc, char **argv, void *userdata) {
         if (r < 0)
                 return log_oom();
 
+        if (arg_root) {
+                if (streq(arg_debugger, "gdb")) {
+                        const char *sysroot_cmd, *src_dir_cmd;
+                        sysroot_cmd = strjoina("set sysroot ", arg_root);
+                        src_dir_cmd = strjoina("dir ", arg_root);
+
+                        r = strv_extend_strv(&debugger_call, STRV_MAKE("-iex", sysroot_cmd, "-iex", src_dir_cmd), false);
+                        if (r < 0)
+                                return log_oom();
+                } else if (streq(arg_debugger, "lldb")) {
+                        _cleanup_free_ char *mapped_exe_dir = strdup(exe_dir);
```

Comment:
Missing failure check here. Also, we tend to separate declarations from assigments involving logic, so I'd split this into the declaration followed by the strdup().